### ISSUE TITLE
i cant build bsnes anymore in arch linux 

### ISSUE THREAD

**Issue by amayra** [bsnes-emu/bsnes#308]
i get this error : 
```
Preparing...
Cloning bsnes build files...
Generating bsnes information...
Checking bsnes dependencies...
Resolving dependencies...
Checking inter-conflicts...

Building bsnes...
==> Making package: bsnes 115-2 (Tue Dec 12 01:45:07 2023)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Found v115.tar.gz
  -> Found package.patch
==> Validating source files with sha256sums...
    v115.tar.gz ... Passed
    package.patch ... Passed
==> Removing existing $srcdir/ directory...
==> Extracting sources...
  -> Extracting v115.tar.gz with bsdtar
bsdtar: Failed to set default locale
==> Starting prepare()...
patching file bsnes/GNUmakefile
Hunk #1 succeeded at 1 with fuzz 2.
patching file bsnes/target-bsnes/GNUmakefile
==> Removing existing $pkgdir/ directory...
==> Starting build()...
make: Entering directory '/var/tmp/pamac-build-rymaa/bsnes/src/bsnes-115/bsnes'
Compiling ../hiro/hiro.cpp ...
Compiling ../ruby/ruby.cpp ...
Compiling ../libco/libco.c ...
In file included from /usr/include/gtksourceview-2.0/gtksourceview/gtksourcebuffer.h:32,
                 from /usr/include/gtksourceview-2.0/gtksourceview/gtksourceview.h:29,
                 from ../hiro/core/../gtk/header.hpp:41,
                 from ../hiro/core/core.cpp:8,
                 from ../hiro/hiro.cpp:5:
/usr/include/gtksourceview-2.0/gtksourceview/gtksourcestylescheme.h:64:13: warning: Deprecated pre-processor symbol: replace with "const"
   64 | G_CONST_RETURN gchar* G_CONST_RETURN *
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~                               
/usr/include/gtksourceview-2.0/gtksourceview/gtksourcestylescheme.h:64:13: warning: Deprecated pre-processor symbol: replace with "const"
In file included from ../hiro/core/../gtk/header.hpp:42:
/usr/include/gtksourceview-2.0/gtksourceview/gtksourcelanguagemanager.h:65:13: warning: Deprecated pre-processor symbol: replace with "const"
   65 | G_CONST_RETURN gchar* G_CONST_RETURN *
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~                               
/usr/include/gtksourceview-2.0/gtksourceview/gtksourcelanguagemanager.h:65:13: warning: Deprecated pre-processor symbol: replace with "const"
/usr/include/gtksourceview-2.0/gtksourceview/gtksourcelanguagemanager.h:71:13: warning: Deprecated pre-processor symbol: replace with "const"
   71 | G_CONST_RETURN gchar* G_CONST_RETURN *
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~                               
/usr/include/gtksourceview-2.0/gtksourceview/gtksourcelanguagemanager.h:71:13: warning: Deprecated pre-processor symbol: replace with "const"
In file included from ../hiro/core/../gtk/header.hpp:43:
/usr/include/gtksourceview-2.0/gtksourceview/gtksourcestyleschememanager.h:76:13: warning: Deprecated pre-processor symbol: replace with "const"
   76 | G_CONST_RETURN gchar* G_CONST_RETURN *
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~                               
/usr/include/gtksourceview-2.0/gtksourceview/gtksourcestyleschememanager.h:76:13: warning: Deprecated pre-processor symbol: replace with "const"
/usr/include/gtksourceview-2.0/gtksourceview/gtksourcestyleschememanager.h:81:13: warning: Deprecated pre-processor symbol: replace with "const"
   81 | G_CONST_RETURN gchar* G_CONST_RETURN *
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~                               
/usr/include/gtksourceview-2.0/gtksourceview/gtksourcestyleschememanager.h:81:13: warning: Deprecated pre-processor symbol: replace with "const"
In file included from /usr/include/gtk-2.0/gtk/gtkobject.h:37,
                 from /usr/include/gtk-2.0/gtk/gtkwidget.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkcontainer.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkbin.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkwindow.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkdialog.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkaboutdialog.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:33,
                 from ../hiro/core/../gtk/header.hpp:38:
/usr/include/gtk-2.0/gtk/gtktypeutils.h:236:64: warning: 'GTypeDebugFlags' is deprecated [-Wdeprecated-declarations]
  236 | void            gtk_type_init   (GTypeDebugFlags    debug_flags);
      |                                                                ^
In file included from /usr/include/glib-2.0/gobject/gobject.h:26,
                 from /usr/include/glib-2.0/gobject/gbinding.h:31,
                 from /usr/include/glib-2.0/glib-object.h:24,
                 from /usr/include/glib-2.0/gio/gioenums.h:30,
                 from /usr/include/glib-2.0/gio/giotypes.h:30,
                 from /usr/include/glib-2.0/gio/gio.h:28,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from ../hiro/core/../gtk/header.hpp:35:
/usr/include/glib-2.0/gobject/gtype.h:729:3: note: declared here
  729 | } GTypeDebugFlags GOBJECT_DEPRECATED_TYPE_IN_2_36;
      |   ^~~~~~~~~~~~~~~
In file included from /usr/include/gtk-2.0/gtk/gtktoolitem.h:31,
                 from /usr/include/gtk-2.0/gtk/gtktoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtkmenutoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtk.h:126:
/usr/include/gtk-2.0/gtk/gtktooltips.h:73:12: warning: 'GTimeVal' is deprecated: Use 'GDateTime' instead [-Wdeprecated-declarations]
   73 |   GTimeVal last_popdown;
      |            ^~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib/galloca.h:34,
                 from /usr/include/glib-2.0/glib.h:32,
                 from /usr/include/glib-2.0/gobject/gbinding.h:30:
/usr/include/glib-2.0/glib/gtypes.h:578:26: note: declared here
  578 | typedef struct _GTimeVal GTimeVal GLIB_DEPRECATED_TYPE_IN_2_62_FOR(GDateTime);
      |                          ^~~~~~~~
In file included from ../nall/arithmetic.hpp:37,
                 from ../nall/hash/hash.hpp:3,
                 from ../nall/hash/sha256.hpp:3,
                 from ../nall/file-buffer.hpp:12,
                 from ../nall/file.hpp:3,
                 from ../nall/directory.hpp:3,
                 from ../ruby/ruby.hpp:4,
                 from ../ruby/ruby.cpp:1:
../nall/arithmetic/natural.hpp: In function 'void nall::div(const uint256_t&, const uint256_t&, uint256_t&, uint256_t&)':
../nall/arithmetic/natural.hpp:239:23: error: 'runtime_error' is not a member of 'std'
  239 |   if(!rhs) throw std::runtime_error("division by zero");
      |                       ^~~~~~~~~~~~~
../nall/arithmetic/natural.hpp:1:1: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
  +++ |+#include <stdexcept>
    1 | #define ConcatenateType(Size) uint##Size##_t
In file included from ../nall/arithmetic.hpp:45:
../nall/arithmetic/natural.hpp: In function 'void nall::div(const uint512_t&, const uint512_t&, uint512_t&, uint512_t&)':
../nall/arithmetic/natural.hpp:239:23: error: 'runtime_error' is not a member of 'std'
  239 |   if(!rhs) throw std::runtime_error("division by zero");
      |                       ^~~~~~~~~~~~~
../nall/arithmetic/natural.hpp:239:23: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
In file included from ../nall/arithmetic.hpp:53:
../nall/arithmetic/natural.hpp: In function 'void nall::div(const uint1024_t&, const uint1024_t&, uint1024_t&, uint1024_t&)':
../nall/arithmetic/natural.hpp:239:23: error: 'runtime_error' is not a member of 'std'
  239 |   if(!rhs) throw std::runtime_error("division by zero");
      |                       ^~~~~~~~~~~~~
../nall/arithmetic/natural.hpp:239:23: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
In file included from ../nall/arithmetic.hpp:61:
../nall/arithmetic/natural.hpp: In function 'void nall::div(const uint2048_t&, const uint2048_t&, uint2048_t&, uint2048_t&)':
../nall/arithmetic/natural.hpp:239:23: error: 'runtime_error' is not a member of 'std'
  239 |   if(!rhs) throw std::runtime_error("division by zero");
      |                       ^~~~~~~~~~~~~
../nall/arithmetic/natural.hpp:239:23: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
In file included from ../nall/arithmetic.hpp:69:
../nall/arithmetic/natural.hpp: In function 'void nall::div(const uint4096_t&, const uint4096_t&, uint4096_t&, uint4096_t&)':
../nall/arithmetic/natural.hpp:239:23: error: 'runtime_error' is not a member of 'std'
  239 |   if(!rhs) throw std::runtime_error("division by zero");
      |                       ^~~~~~~~~~~~~
../nall/arithmetic/natural.hpp:239:23: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
In file included from ../nall/arithmetic.hpp:77:
../nall/arithmetic/natural.hpp: In function 'void nall::div(const uint8192_t&, const uint8192_t&, uint8192_t&, uint8192_t&)':
../nall/arithmetic/natural.hpp:239:23: error: 'runtime_error' is not a member of 'std'
  239 |   if(!rhs) throw std::runtime_error("division by zero");
      |                       ^~~~~~~~~~~~~
../nall/arithmetic/natural.hpp:239:23: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
In file included from ../nall/arithmetic.hpp:37,
                 from ../nall/hash/hash.hpp:3,
                 from ../nall/hash/sha256.hpp:3,
                 from ../nall/file-buffer.hpp:12,
                 from ../nall/file.hpp:3,
                 from ../nall/directory.hpp:3,
                 from ../hiro/core/core.hpp:4,
                 from ../hiro/core/core.cpp:13:
../nall/arithmetic/natural.hpp: In function 'void nall::div(const uint256_t&, const uint256_t&, uint256_t&, uint256_t&)':
../nall/arithmetic/natural.hpp:239:23: error: 'runtime_error' is not a member of 'std'
  239 |   if(!rhs) throw std::runtime_error("division by zero");
      |                       ^~~~~~~~~~~~~
../nall/arithmetic/natural.hpp:1:1: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
  +++ |+#include <stdexcept>
    1 | #define ConcatenateType(Size) uint##Size##_t
In file included from ../nall/arithmetic.hpp:45:
../nall/arithmetic/natural.hpp: In function 'void nall::div(const uint512_t&, const uint512_t&, uint512_t&, uint512_t&)':
../nall/arithmetic/natural.hpp:239:23: error: 'runtime_error' is not a member of 'std'
  239 |   if(!rhs) throw std::runtime_error("division by zero");
      |                       ^~~~~~~~~~~~~
../nall/arithmetic/natural.hpp:239:23: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
In file included from ../nall/arithmetic.hpp:53:
../nall/arithmetic/natural.hpp: In function 'void nall::div(const uint1024_t&, const uint1024_t&, uint1024_t&, uint1024_t&)':
../nall/arithmetic/natural.hpp:239:23: error: 'runtime_error' is not a member of 'std'
  239 |   if(!rhs) throw std::runtime_error("division by zero");
      |                       ^~~~~~~~~~~~~
../nall/arithmetic/natural.hpp:239:23: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
In file included from ../nall/arithmetic.hpp:61:
../nall/arithmetic/natural.hpp: In function 'void nall::div(const uint2048_t&, const uint2048_t&, uint2048_t&, uint2048_t&)':
../nall/arithmetic/natural.hpp:239:23: error: 'runtime_error' is not a member of 'std'
  239 |   if(!rhs) throw std::runtime_error("division by zero");
      |                       ^~~~~~~~~~~~~
../nall/arithmetic/natural.hpp:239:23: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
In file included from ../nall/arithmetic.hpp:69:
../nall/arithmetic/natural.hpp: In function 'void nall::div(const uint4096_t&, const uint4096_t&, uint4096_t&, uint4096_t&)':
../nall/arithmetic/natural.hpp:239:23: error: 'runtime_error' is not a member of 'std'
  239 |   if(!rhs) throw std::runtime_error("division by zero");
      |                       ^~~~~~~~~~~~~
../nall/arithmetic/natural.hpp:239:23: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
In file included from ../nall/arithmetic.hpp:77:
../nall/arithmetic/natural.hpp: In function 'void nall::div(const uint8192_t&, const uint8192_t&, uint8192_t&, uint8192_t&)':
../nall/arithmetic/natural.hpp:239:23: error: 'runtime_error' is not a member of 'std'
  239 |   if(!rhs) throw std::runtime_error("division by zero");
      |                       ^~~~~~~~~~~~~
../nall/arithmetic/natural.hpp:239:23: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
make: *** [../ruby/GNUmakefile:76: obj/ruby.o] Error 1
make: *** Waiting for unfinished jobs....
make: *** [../hiro/GNUmakefile:73: obj/hiro-gtk2.o] Error 1
make: Leaving directory '/var/tmp/pamac-build-rymaa/bsnes/src/bsnes-115/bsnes'
==> ERROR: A failure occurred in build().
    Aborting...
```

**Comment by Screwtapello** [comment]
bsnes v115 is incompatible with modern compilers, if Arch has a "bsnes-git" package that will probably work better, or you could just build it from source yourself.

**Comment by orbea** [comment]
@Screwtapello Are there any plans to release bug fix releases?

**Comment by erus** [comment]
The broken bsnes package has been [removed from the AUR](https://lists.archlinux.org/archives/list/aur-requests@lists.archlinux.org/message/JXSAYI7DNUO64VS3ZWYP3COKVTQ64ZAE/), this issue can probably be safely closed. For anyone on Arch, there is still [bsnes-qt5](https://aur.archlinux.org/packages/bsnes-qt5) available, building on commit 7ec3077079445c1ab591d542d475da5fa121632e.
